src/java.base/share/classes/java/security/Security.java

```diff
@@ -141,8 +141,10 @@ private static void initialize() {
         }
 
         /*[IF CRIU_SUPPORT]*/
-        // Check if CRIU checkpoint mode is enabled, if it is then reconfigure the security providers.
-        if (InternalCRIUSupport.isCheckpointAllowed()) {
+        // Check if CRIU checkpoint mode is enabled.
+        // Check if the -XX:-CRIUSecProvider JVM option is enabled, if it is not, then use the CRIU security provider.
+        // Otherwise, use the normal providers from the java.security file.
```

Comment:
The code doesn't match these comments; I suggest you simply remove them.

Reply:
Removed